File: c++/3005.cpp
// Link: https://leetcode.com/problems/count-elements-with-maximum-frequency/description/
// Runtime: 7 ms
// Memory: 22.92 MB

class Solution 
{
public:
    int maxFrequencyElements(vector<int>& nums) 
    {
        std::unordered_map<int, int> elementAndFrequency;
        int count = 0, maxFrequency = 0;
        for (const auto& val : nums)
        {
            elementAndFrequency[val]++;

            if (elementAndFrequency[val] > maxFrequency)
            {
                maxFrequency = elementAndFrequency[val];
                count = 1;
            }
            else if (elementAndFrequency[val] == maxFrequency)
                count++;
        }

        return count * maxFrequency;
    }
};

File: python/3005.py
# Link: https://leetcode.com/problems/count-elements-with-maximum-frequency/description/
# Runtime: 51 ms
# Memory: 16.70 MB


class Solution:
    def maxFrequencyElements(self, nums: List[int]) -> int:
        element_and_frequency = Counter(nums).most_common()
        max_frequency, i = element_and_frequency[0][1], 1
        while (i < len(element_and_frequency)) and element_and_frequency[i][
            1
        ] == element_and_frequency[i - 1][1]:
            max_frequency += element_and_frequency[i][1]
            i += 1

        return max_frequency
